spec/factories/course_discussion_posts.rb

```diff
@@ -1,10 +1,22 @@
 # frozen_string_literal: true
 FactoryGirl.define do
   factory :course_discussion_post, class: Course::Discussion::Post.name do
+    transient do
+      vote_count 0
+      user nil
+    end
+
     creator
     updater
     parent nil
     association :topic, factory: :course_discussion_topic
     text 'This is a test post'
+
+    after(:create) do |post, evaluator|
+      next if evaluator.user.blank? || evaluator.vote_count == 0
+      evaluator.vote_count.times do
+        post.cast_vote!(evaluator.user, 1)
```

Comment:
If `vote_count` is 1, and `user` is `nil`, will that be an issue? Might need to enumerate through the `user` attribute. 

Also, I read through the `#cast_vote` method - if we call the same method twice with the same user, it actually only creates 1 upvote. 

I'm thinking if there are other ways to define this transient attribute - possibly having `upvoters` and `downvoters` instead, each taking in an array of users - what do you think? Or if you have other ideas, you can suggest too!

Reply:
You're right, it'll only create a single vote, which wasn't what I intended it to be.

I thought of using a Hash like structure at first, but the `upvoters` and `downvoters` approach was cleaner. I've updated it accordingly.